src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -147,8 +273,41 @@ const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({ submiss
     );
   }, [columnFilters, globalFilter, dispatch]);
 
+  // tells page to ask for new entries from main page when its state changes.
+  useEffect(() => {
+    updateEntries(limitGroup, pageParams, backendFilterParams);
+    // updateEntries seems to change between renders as a reference?
+    // eslint-disable-next-line
+  }, [limitGroup, pageParams, backendFilterParams]);
+
   return (
     <>
+      <Flex justifyContent="justify-start" marginTop="mt-2" spaceX="space-x-2">
+        <Text>Viewing</Text>
+        <SimpleDropdown
+          options={showSubmissionOptions}
+          selectedValue={showAllSubmissions}
+          onClick={setStateAndReset(setShowAllSubmissions)}
+          popoverProps={{ position: Position.BOTTOM }}
+          buttonProps={{ minimal: true, rightIcon: 'caret-down' }}
+        />
+        <Text>Submissions from</Text>
+        <SimpleDropdown
+          options={groupOptions}
+          selectedValue={limitGroup}
+          onClick={setStateAndReset(setLimitGroup)}
+          popoverProps={{ position: Position.BOTTOM }}
+          buttonProps={{ minimal: true, rightIcon: 'caret-down' }}
+        />
```

Comment:
Yea, I'm in favour of moving these back to `Grading` to keep the logic separate. After all, you're already using dependency injection to pass `updateEntries` as a prop – why can't the value of group: false/true be defined from `Grading`?